### PR TITLE
📦 [Dependencies.swift] Define SWIFT_PACKAGE for SPM targets

### PR DESCRIPTION
**Pull request by danyf90**
### Short description 📝

Some SPM packages checks the SWIFT_PACKAGE definition (e.g. https://github.com/mattgallagher/CwlCatchException/blob/682841464136f8c66e04afe5dbd01ab51a3a56f2/Sources/CwlCatchException/CwlCatchException.swift#L23-L25), so it should be defined for all SPM based targets.

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift
Comment (pepibumur) [inline Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift:407]: It'd be great to add a test here that ensures that we are setting that build setting. The test will prevent regressions in the future.

**Review by pepibumur — APPROVED** [review]
Besides the comment, I'd update the CHANGELOG. After that, I think we are good to merge if CI is green.

File: Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift
Comment (danyf90) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift:407]: It is tested by the test that uses the `Sources/TuistDependenciesTesting/DependenciesGraph/DependenciesGraph+TestData.swift` I have changed in this PR

**Comment by danyf90** [comment]
> Besides the comment, I'd update the CHANGELOG. After that, I think we are good to merge if CI is green.

Given we have never released SPM support yet, I don't t think we should add an entry for this specific case, but it would be part of the generic "SPM support"

**Review by fortmarek — APPROVED** [review]

File: Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift
Comment (pepibumur) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift:407]: Ouch, I see. In that case, I suggest taking a different approach. The fact that you are implicitly describing the test through test data is not obvious for a contributor. I think tests should be very explicit:

```swift
func loaded_targets_have_the_swift_package_compiler_macro() {
  // Test implementation
}
```

As a contributor to the project, if I see that test is failing, I know what might have caused it and can fix it quickly. If I change something in the test data and tests start failing, it'd be hard for me to know what's causing the tests to fail.

**Pull request merged**
